src/jpegreadframe.c

```diff
@@ -208,8 +208,8 @@ void unpack_JPEG_component (double * restrict result, double * restrict source,
     scaleV = 1;
   }
   // indexes into the interpolation index lists: 1-4 for integer ratios (scale = 1 after normalization above), 0 for 3/2, 5 for 4/3
-  unsigned char indexH = (scaleH != 1) ? (scaleH == 3) * 5 : maxH;
-  unsigned char indexV = (scaleV != 1) ? (scaleV == 3) * 5 : maxV;
+  unsigned char indexH = scaleH == 1 ? maxH : scaleH == 3 ? 5 : 0;
```

Comment:
These could be one line; I wouldn't, but that seems like the consistent style here:
```c
  unsigned char indexH = scaleH == 1 ? maxH : scaleH == 3 ? 5 : 0, indexV = scaleV == 1 ? maxV : scaleV == 3 ? 5 : 0;
```

Reply:
Should have parentheses around the conditions for consistency, but other than that, I don't see the problem. Might be more efficient to check for 1 and 2 instead of 1 and 3, perhaps? (The only possible values here are 1, 2 and 3.)

Reply:
I like the sound of that:
```c
  unsigned char indexH = (scaleH == 2) ? 0 : (scaleH == 3) ? 5 : maxH, indexV = (scaleV == 2) ? 0 : (scaleV == 3) ? 5 : maxV;
```